coiote-aws-iot-cloud-formation/certificates/certificates.py

```diff
@@ -0,0 +1,124 @@
+import json
+import os
+
+import boto3
+import requests
+from crhelper import CfnResource
+from OpenSSL import crypto
+
+helper = CfnResource()
+iot_client = boto3.client('iot')
+secrets_manager_client = boto3.client('secretsmanager')
+
+user = os.environ['coioteDMrestUsername']
+password = os.environ['coioteDMrestPassword']
+rest_uri = os.environ['coioteDMrestUri']
+group_id = os.environ['coioteDMrestGroupId']
+
+secret_name = 'coioteDMcert'
+
+
+@helper.create
+def create(event, context):
+
+    internal_certificate = iot_client.create_keys_and_certificate(setAsActive=True)
+    external_certificate = generate_external_cert(email_address=user, common_name=user)
+    ca_result = requests.get('https://www.amazontrust.com/repository/AmazonRootCA1.pem')
+    internal_cert_ca = ca_result.content.decode('UTF-8')
+
+    send_internal_certificate(internal_certificate, internal_cert_ca)
+
+    send_external_certificate(external_certificate)
+
+    try:
+        save_certificate_in_secret_manager(external_certificate)
+    except secrets_manager_client.exceptions.ResourceExistsException:
+        delete_internal_certificate()
+        delete_external_certificate()
```

Comment:
To już chyba lepiej zmienić kolejność, nie trzeba wtedy będzie po błędzie usuwać certów kojota. No i jeszcze powinien być ten cert usuwany z iot core jak już wspomniałeś